Wyam/config.wyam

```diff
@@ -1,16 +1,16 @@
-#n Wyam.Razor
-
-#a "..\Solutions\Endjin.AzureWeekly.Wyam\bin\Debug\netstandard2.0\Endjin.AzureWeekly.WyamModules.dll"
-
-#theme AzureWeekly
-
-// Customize your settings and add new ones here
-Settings[Keys.Host] = "AzureWeekly.info";
-
-string newscollectionFilePath = @"..\Artefacts\NewsletterCollections\168.json";
-
-Pipelines.Add(
-    "Newsletter",
-    NewsTopicModule(newscollectionFilePath),
-    Razor(),
-    WriteFiles());
\ No newline at end of file
+#n Wyam.Razor
+
+#a "..\Solutions\Endjin.AzureWeekly.Wyam\bin\Debug\netstandard2.0\Endjin.AzureWeekly.WyamModules.dll"
+
+#theme AzureWeekly
+
+// Customize your settings and add new ones here
+Settings[Keys.Host] = "AzureWeekly.info";
+
+string newscollectionFilePath = @"..\Artefacts\NewsletterCollections\168.json";
+
+Pipelines.Add(
+    "Newsletter",
+    NewsTopicModule(newscollectionFilePath),

```

Comment:
Nice work on writing a module! You were _so_ close to getting everything wired up - I'll explain my changes to the next few lines inline. With regard to the module, putting it in a separate assembly is totally fine and works well, especially when the module needs other supporting classes. Just in case you weren't aware, you can also put the code for new modules directly in your config file (I usually add them at the bottom).

The important thing to note is that at the conclusion of this module you have a collection of `IDocument` objects, each with metadata keys like "NewsletterFrom" and _empty_ content. In this example code, you're only outputting one `IDocument` right now, but it looks like the intent is for `NewsTopicModule` to output several documents.

Reply:
I did h\ve it inline first time around, but moved it into a project do that I could hit Debugger.Launch() to see what was going on...

I love how you've  build a C#-y DSL - I need to look at the source and figure out how you've done that. Last time I tried doing something similar, it was with Boo and it was quite painful.

Reply:
> I love how you've build a C#-y DSL

Umm...about that 🤣 It's totally going away in the next major version (v3). Instead "custom config" will be done entirely via a normal .NET project and bootstrapped as a console app. That's mostly in reaction to lots of feedback regarding lack of Intellisense and debugging support as well as a desire to limit the amount of Roslyn/NuGet maintenance I have to do (though I do love Roslyn). Making everything build via the normal .NET project system also has some advantages over a runtime config file and dynamic assembly loading that will make life easier on me.

For more information, see this issue: https://github.com/Wyamio/Wyam/issues/668

And here's an early look at what bootstrapping your custom Wyam generator might look like:
![D1-gwL9UwAENKJ6](https://user-images.githubusercontent.com/1020407/55094798-78def800-508d-11e9-8181-a7728aacc3eb.png)

Reply:
I think that's the right approach - it does make life a lot simpler for all the reasons you've outlined. I also like the fact that you can either use as a .NET Global Tool or to integrate into your own tool / platform. I've been thinking about how I integrate this directly into the content platform we've build for the Azure Weekly Newsletter - directly outputting into Azure Blob Storage or some such...